onnxruntime/core/providers/common.h

```diff
@@ -132,4 +135,14 @@ inline Status ComputePadAndOutputShape(const int64_t in_dim,
   return Status::OK();
 }
 
+template <class Map, class Key>
+inline bool Contains(const Map& map, const Key& key) {
+  return map.find(key) != map.end();
+}
+
+template <template <typename...> class Container, typename T>
+T Product(const Container<T>& c) {
+  return static_cast<T>(accumulate(c.cbegin(), c.cend(), 1, std::multiplies<T>()));
+}
```

Comment:
Should maybe use SafeInt for this operation to make sure there's no overflow.

Reply:
accumulate and multiplies will both return T, so if there is overflow, it will happen inside 
Added a comment says this will not have overflow protection

---
In reply to: [566657933](https://github.com/microsoft/onnxruntime/pull/6498#discussion_r566657933) [](ancestors = 566657933)